src/Model/DataModel/StreamData/HoldEmailData.php

```diff
@@ -40,19 +45,48 @@ class HoldEmailData extends StreamData
     public $deliveryLocation = '';
 
     /**
+     * @var bool
+     */
+    public $success;
+
+    /**
      * @param Patron $patron
      * @param Bib $bib
      * @param Item $item
      * @param HoldRequest $holdRequest
+     * @param HoldRequestResult $holdRequestResult
      */
-    public function assembleData(Patron $patron, Bib $bib, Item $item, HoldRequest $holdRequest)
+    public function assembleData(Patron $patron, Bib $bib, Item $item, HoldRequest $holdRequest, HoldRequestResult $holdRequestResult)
     {
         $this->setAuthor($bib->getAuthor());
         $this->setBarcode($item->getBarcode());
-        $this->setPatronName($patron->getNames()[0]);
         $this->setTitle($bib->getTitle());
         $this->setDeliveryLocation($holdRequest->getDeliveryLocation());
         $this->setPickupLocation($holdRequest->getPickupLocation());
+        $this->setSuccess($holdRequestResult->isSuccess());
+
+        $name = $patron->getNames()[0];
+        $fullName = explode(",", $name);
+        $fullName[1] = trim($fullName[1]);
+        $name = ucfirst(strtolower($fullName[1])) . " " . ucfirst(strtolower($fullName[0]));
+        $this->setPatronName($name);
+
+        /**
+         * @var DocDeliveryData
+         */
+        $docDeliveryData = $holdRequest->getDocDeliveryData();
+        $email = '';
+
+        if ($docDeliveryData !== null) {
+            $email = $docDeliveryData->getEmailAddress();
+        }
+
+        // If request is not an EDD, use e-mail from patron's info.
+        if ($email !== '') {
+            $this->setPatronEmail($email);
+        } else if (count($patron->getEmails()) > 0){
```

Comment:
@emu47 This should only set e-mail from Patron Info when e-mail is found in the Patron object.